### PR TITLE
Update contributing docs for new dev server

### PR DESCRIPTION
**Pull request by studioswong**
## Description

related ticket: https://jira.quantumblack.com/browse/KED-2344

This PR is mainly to test and update the contributing docs to provide more clarity after I have tried out the workflow with development using the newly implemented kedro dev server command. 

## Development notes

I have updated the contributing docs for more clarity. 

## QA notes

N/A

## Checklist

- [x] Read the [contributing](/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added new entries to the `RELEASE.md` file
- [ ] Added tests to cover my changes

## Legal notice

- [ ] I acknowledge and agree that, by checking this box and clicking "Submit Pull Request":

- I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
- I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorised to submit this contribution on behalf of the original creator(s) or their licensees.
- I certify that the use of this contribution as authorised by the Apache 2.0 license does not violate the intellectual property rights of anyone else.

### CODE REVIEW

**Comment by studioswong** [comment]
py tests are failing at the moment - will check with core team to see how they could be fixed.

File: CONTRIBUTING.md
Comment (richardwestenra) [inline CONTRIBUTING.md:90]: > [localhost:4141]

This should be 

```
[localhost:4141](http://localhost:4141)
```

Comment (richardwestenra) [inline CONTRIBUTING.md:90]: Wording/formatting suggestion:

```suggestion
This command will launch a Kedro-Viz server at [localhost:4142](http://localhost:4142) and serve data from a real Kedro pipeline located at the project path supplied to the command. From then on, launching the app locally at [localhost:4141](http://localhost:4141) will pull data from the Kedro-Viz server that is running on port 4142.

Note: Once the development server is launched at port 4142, the local app will always pull data from that server. To prevent this, you can comment out the proxy setting in `package.json` and relaunch your app.
```

Comment (richardwestenra) [reply CONTRIBUTING.md:90]: > relaunch your app

By this, do you mean restart the dev server at 4141? If so, I'd say that instead.

**Review by richardwestenra — CHANGES_REQUESTED** [review]
This is great, thanks! I've added a couple of small suggestions (the URL one is the only one that 100% needs changing). I also suggest that you ask @stichbury to look at it if she's around.

File: CONTRIBUTING.md
Comment (studioswong) [reply CONTRIBUTING.md:90]: sounds good, I'll state that instead.

**Review by richardwestenra — APPROVED** [review]
Thanks! LGTM 👍

I still think it'd be good for @stichbury to take a look before this gets merged if possible

File: CONTRIBUTING.md
Comment (stichbury) [inline CONTRIBUTING.md:90]:
```suggestion
> **Note**: Once the development server is launched at port 4142, the local app will always pull data from that server. To prevent this, you can comment out the proxy setting in `package.json` and restart the dev server at port 4141.
```

Comment (stichbury) [reply CONTRIBUTING.md:90]: Just adding some formatting to your note.

**Review by stichbury — APPROVED** [review]
Looks great to me 🥇 -- just one minor formatting suggestion.

File: CONTRIBUTING.md
Comment (studioswong) [reply CONTRIBUTING.md:90]: Thank you @stichbury ! I'll add that in.

Comment (studioswong) [reply CONTRIBUTING.md:90]: on going through the notes again while adding the suggested formatting, I noticed on line 42 of the docs that we use `> _Note:_` to format notes instead of `> **Note**` - my guess is to follow the previous formatting? ( I could also change the formatting within the entire document to follow `> **Note**` instead as well.) @stichbury WDYT?

Comment (richardwestenra) [reply CONTRIBUTING.md:90]: I agree, bold is probably better than italics for this, and I'd change the other one to use bold as well, for consistency's sake

**Pull request merged**
